src/core/classification/qgsclassificationmethod.h

```diff
@@ -283,20 +302,32 @@ class CORE_EXPORT QgsClassificationMethod SIP_ABSTRACT
     //! Format the number according to label properties
     QString formatNumber( double value ) const;
 
+    /**
+     * Add parameter
+     * \since QGIS 3.12
+     */
+    void addParameter( QgsProcessingParameterDefinition *definition );
```

Comment:
```suggestion
    void addParameter( QgsProcessingParameterDefinition *definition SIP_TRANSFER );
```

Dox could use a lot of work on this one :wink:

Reply:
Better also note in the dox that only certain parameter types are supported -- i.e. those which have had their widgets ported to c++.

Pure python widgets obviously won't work here.